gems/sorbet-runtime/lib/types/types/typed_enumerator.rb

```diff
@@ -18,5 +18,15 @@ def valid?(obj)
     def new(*args, &blk) # rubocop:disable PrisonGuard/BanBuiltinMethodOverride
       T.unsafe(Enumerator).new(*args, &blk)
     end
+
+    class Untyped < TypedEnumerator
```

Comment:
I wanted to preserve the types for inspection by metaprogramming and their ilk; Basically, this felt like it moves us towards the invariant of "no bare generic types" in the runtime type system, instead of moving us away from it.

I was also specifically under the impression that when @zanker-stripe tried the optimization of making these `Simple` he ran into some problems where other locations that were inspecting these types (in e.g. Thrift codegen maybe?) were unhappy with seeing Simple types instead of the typed ones. I may have misunderstood the problems, though.